src/fivetran-destination/src/destination/dml.rs

```diff
@@ -417,16 +421,27 @@ async fn delete_files(
     // TODO(parkmycar): Make sure table exists.
     // TODO(parkmycar): Retry transient errors.
 
-    // Copy into a temporary table, which we then merge into the destination.
-    let ts = SystemTime::now()
-        .duration_since(SystemTime::UNIX_EPOCH)
-        .context("time went backwards!")?;
-    // TODO(parkmycar): Make sure table name is <= 255 characters.
-    let temp_table_name = format!("fivetran_{}_temp_delete_{}", table.name, ts.as_millis());
+    // Bail early if there is no work to do.
+    if delete_files.is_empty() {
+        return Ok(());
+    }
 
-    let schema_name = escape::escape_identifier(schema);
-    let table_name = escape::escape_identifier(&table.name);
-    let temp_table_name = escape::escape_identifier(&temp_table_name);
+    // Copy into a temporary table, which we then merge into the destination.
+    //
+    // This temporary table names to be a valid identifier within Materialize, and a table with the
+    // same name must not already exist in the provided schema.
+    let prefix = format!("fivetran_temp_{}_", mz_ore::id_gen::temp_id());
+    let mut temp_table_name = Ident::new_lossy(prefix);
+    temp_table_name.append_lossy(&table.name);
+    let temp_table_name = temp_table_name.into_string();
+
+    let qualified_table_name = format!(
```

Comment:
If you are using `Ident` already, does it make sense to use `UnresolvedItemName` here? I kinda don't care, but it avoids manual string formatting which, for sql idents, always scares me, even with the escaping.

Reply:
Good idea, changed to use `UnresolvedItemName`!